jme3-examples/src/main/java/jme3test/terrain/PBRTerrainAdvancedTest.java

```diff
@@ -0,0 +1,429 @@
+    private float marbleScale = 64;
+    private float gravelScale = 64;
+    
+    private AmbientLight ambientLight;
+    private DirectionalLight directionalLight;
+    private boolean isNight = false;
+    private final float dayLightIntensity = 1.0f;
+    private final float nightLightIntensity = 0.03f;
+    
+    private BitmapText keybindingsText;
+
+    public static void main(String[] args) {
+        PBRTerrainAdvancedTest app = new PBRTerrainAdvancedTest();
+        AppSettings s = new AppSettings(true);
+        
+        s.put("GammaCorrection", true);
+        
+        app.setSettings(s);
+        app.start();
+        
+ 
+       
+    }
+    
+    private final ActionListener actionListener = new ActionListener() {
+        @Override
+        public void onAction(String name, boolean pressed, float tpf) {
+            if (name.equals("triPlanar") && !pressed) {
+                triPlanar = !triPlanar;
+                if (triPlanar) {
+                    matTerrain.setBoolean("useTriPlanarMapping", true);
+                    // planar textures don't use the mesh's texture coordinates but real world coordinates,
+                    // so we need to convert these texture coordinate scales into real world scales so it looks
+                    // the same when we switch to/from tr-planar mode (1024f is the alphamap size)
+                    matTerrain.setFloat("AlbedoMap_0_scale", 1f /  (1024f / dirtScale));
+                    matTerrain.setFloat("AlbedoMap_1_scale", 1f /  (1024f / darkRockScale));
+                    matTerrain.setFloat("AlbedoMap_2_scale", 1f /  (1024f / snowScale));
+                    matTerrain.setFloat("AlbedoMap_3_scale", 1f /  (1024f / tileRoadScale));
+                    matTerrain.setFloat("AlbedoMap_4_scale", 1f /  (1024f / grassScale));
+                    matTerrain.setFloat("AlbedoMap_5_scale", 1f /  (1024f / marbleScale));
+                    matTerrain.setFloat("AlbedoMap_6_scale", 1f /  (1024f / gravelScale));
+                } else {
+                    matTerrain.setBoolean("useTriPlanarMapping", false);
+
+                    matTerrain.setFloat("AlbedoMap_0_scale", dirtScale);
+                    matTerrain.setFloat("AlbedoMap_1_scale", darkRockScale);
+                    matTerrain.setFloat("AlbedoMap_2_scale", snowScale);
+                    matTerrain.setFloat("AlbedoMap_3_scale", tileRoadScale);
+                    matTerrain.setFloat("AlbedoMap_4_scale", grassScale);
+                    matTerrain.setFloat("AlbedoMap_5_scale", marbleScale);
+                    matTerrain.setFloat("AlbedoMap_6_scale", gravelScale);
+
+                }
+            }
+            if (name.equals("toggleNight") && !pressed) {
+                isNight = !isNight;
+                //ambient and direcitonal light are faded smoothly in update loop below !
+                
+            }
+        }
+    };
+    
+    
+    
+    @Override
+    public void simpleInitApp() {
```

Comment:
The `simpleInitApp()` method is too long to be easily understood. Please break it up into smaller methods.